pkg/serviceactivities/shutdown/shutdownsignal.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: This file holds the ShutdownFromSignalError and Signal structs/enums for use by the ShutdownError
+package shutdown
+
+import "fmt"
+
+// Signal is an enum for which signal type caused the shutdown
+type Signal int
+
+// String satisfies the Stringer interface to convert enum types to strings
+func (s Signal) String() string {
+	switch s {
+	case SignalInterrupt:
+		return "Interrupt"
+	case SignalTerminated:
+		return "Terminated"
+	case SignalHangUp:
+		return "HangUp"
+	}
+	return fmt.Sprintf("Error: %d", s)
+}
+
+// List of signals for thue ShutdownFromSignalError
+const (
+	// SignalInterrupt is SIGINT
```

Comment:
I'm not a huge fan of duplicating this since this is sorta just a copy of the `signal` package.  Can't we just take `os.Signal` as the `Signal` value here?

Reply:
All of the os structures are platform-specific, so it feels bad to rely on things inside the OS-specific files. If you look into it in VSC, it goes into os_posix.go for example.

Reply:
Totally, but I don't _think_ we're at the point where this matters that much. Right? We're already hard depending on those specific syscalls right now, anyways.

I could see the argument for an interface ontop of it later, if we started doing more complicated things.

Reply:
we can try it for now and see if we get compile errors somewhere in the system

Reply:
actually we can't because os.signal only has term and kill

Reply:
oh i see what you're proposing.  yeah the strings will be assy for other signals but probably good enough, lemme see if that works e2e.